src/wallet/memo.h

```diff
@@ -49,32 +49,7 @@ class Memo {
         return result;
     }
 
-    static Memo FromHexOrThrow(const std::string& memoHex) {
-        return std::visit(match {
-            [&](Memo memo) {
-                return memo;
-            },
-            [&](MemoError err) {
-                switch (err) {
-                    case MemoError::HexDecodeError:
-                        throw std::runtime_error(
-                                "Invalid parameter, expected memo data in hexadecimal format.");
-                    case MemoError::MemoTooLong:
-                        throw std::runtime_error(strprintf(
-                                "Invalid parameter, memo is longer than the maximum allowed %d characters.",
-                                ZC_MEMO_SIZE));
-                    default:
-                        assert(false);
-                }
-                // unreachable, but the compiler can't tell
-                return Memo::NoMemo();
-            }
-        }, Memo::FromHex(memoHex));
-    }
-
     // This copies, because if it returns a reference to the underlying value,
```

Comment:
This leaves an incomplete sentence.

Reply:
I actually have a different approach for this PR that does more encapsulation and changes a lot more of this … but I’m afraid it may run afoul of Bitcoin backporting, so it’s sort of in limbo until we have that discussion. Depending on how that goes, it may land closer to where this PR is … which is why this is still in draft.